superbit_lensing/shear_profiles/shear_plots.py

```diff
@@ -26,77 +49,79 @@ def __init__(self, cat_file, truth_file, pix_scale=0.144):
-            return angular_radius * 60.
+            return angular_radius / 60.
         else:
             # in arcsec
             return angular_radius
 
-    def plot_tan_profile(self, title=None, size=(10,7),
-                         label='annular', show=False, outfile=None):
+    def plot_tan_profile(self, title=None, size=(10,7), label='annular',
+                         rbounds=(5, 750), show=False, outfile=None,
+                         nfw_label=None, smoothing=False, plot_truth=True,
+                         xlim=None, ylim=None):
+        '''
+        xlim/ylim: list of tuples
+            A list of len 2 containing the xlim/ylim boundaries for both plots;
+            e.g. ylim=[(-1,5), (-2,2)]
+        '''
+
         rc('font', **{'family':'serif'})
-        rc('text', usetex=True)
-        plt.ion()
-
-
-        radius = get_angular_radius(self.cat['r'], arcmin=True)
-        gtan = self.cat['gtan']
-        gcross = self.cat['gcross']
-        gtan_err = self.cat['gtan_err']
-        gcross_err = self.cat['gcross_err']
-
-        try:
-            data.sort('col1') # get in descending order
-            ## uncomment below if it's needed
-            #data.remove_rows([0])
-            radius=data['col1']
-            radius=radius*pixscale/60.
-        except ValueError:
-            data.sort('r') # get in descending order
-            ## uncomment below if it's needed
-            #data.remove_rows([0])
-            radius=data['r']
-            radius=radius*pixscale/60.
-
-        try:
-            etan=data['gtan']
-            ecross=data['gcross']
-            shear1err=data['err_gtan']
-            shear2err=data['err_gcross']
-
-        except KeyError:
-
-            etan=data['col3']
-            ecross=data['col4']
-            shear1err=data['col5']
-            shear2err=data['col6']
-
-        # So far, this is looking great!!! Now, let's remember how to make bin width error bars
-        minrad = minrad*pixscale/60 #pixels --> arcmin
-        maxrad = maxrad*pixscale/60 #pixels --> arcmin
-
-        upper_err=np.zeros_like(radius)
-        for e in range(len(radius)-1):
-            this_err=(radius[e+1]-radius[e])*0.5
-            upper_err[e]=this_err
-        upper_err[-1]=(maxrad-radius[-1])*0.5
-
-        lower_err=np.zeros_like(radius)
-        for e in (np.arange(len(radius)-1)+1):
-            this_err=(radius[e]-radius[e-1])*0.5
-            lower_err[e]=this_err
-
-        lower_err[0]=(radius[0]-minrad)*0.5
-
-        # And scene
-        rad_err=np.vstack([lower_err,upper_err])
+
+        # used in old plots, won't work unless tex is installed
+        # rc('text', usetex=True)
+        # plt.ion()
+
+        cat = self.cat
+
+        # in arcsec
+        minrad = rbounds[0]
+        maxrad = rbounds[1]
+
+        cat.sort('midpoint_r') # get in descending order
+
+        radius = self.get_angular_radius(cat['midpoint_r'], arcmin=True)
+
+        gtan = cat['gtan_mean']
```

Comment:
I switched the column names to `mean_xxx` and `err_xxx`, I'm sorry. Again, let's merge this in and deal with the column name change afterwards.